server/routes/webhooks/stripe-charge-failed.js

```diff
@@ -0,0 +1,106 @@
+const boom = require('boom');
+const stripe = require('../stripe');
+const basket = require('../../lib/basket-queue');
+
+// values that never change between webhook requests
+const signatureHeader = 'stripe-signature';
+const event_type = 'charge.failed';
+const frequency = 'monthly';
+const recurring = true;
+const service = 'stripe';
+
+// Error/Success message constants
+const forbiddenError = 'An error occurred while verifying the webhook signing secret';
+const incorrectEventError = 'This hook only processes charge failed events';
+const expandChargeError = 'An error occurred while expanding the charge';
+const retrieveSubscriptionErrorMsg = 'An error occurred while fetching the subscription';
+const notRecurringCharge = 'This hook only processes recurring charges that fail';
+const notPartOfSubscription = 'This charge is not part of a subscription';
+const successMessage = 'charge failed event processed';
+
+let stripeChargeFailed = (request, reply) => {
+  // Validate the Webhook signature
+  let endpointSecret = process.env.STRIPE_WEBHOOK_SIGNATURE_CHARGE_FAILED;
+  let signature = request.headers[signatureHeader];
+
+  let event = stripe.constructEvent(request.payload, signature, endpointSecret);
+
+  // If there's no event object, the signature verification failed
+  if (!event) {
+    return reply(boom.forbidden(forbiddenError));
+  }
+
+  // Only process charge.failed webhook events
+  if (event.type !== event_type) {
+    return reply(incorrectEventError);
+  }
+
+  let {charge} = event.data.object;
+
+  // If the charge has no invoice ID, this is not a recurring charge
+  if (!charge.invoice) {
+    return reply(notRecurringCharge);
+  }
+
+  // Expand the invoice object on the charge
+  stripe.retrieveCharge(charge.id, (fetchChargeError, charge) => {
+    if (fetchChargeError) {
+      return reply(boom.badImplementation(`${expandChargeError}: ${fetchChargeError}`));
```

Comment:
one thing to note here is that if `fetchChargeError` is an object and you use string interpolation, it will show up as `[object Object]`

Reply:
I can confirm this in the Stripe code, but I'm almost certain this will be an `Error`, and will have a useful toString implemented for outputting